application/src/main/java/org/togetherjava/tjbot/commands/generic/DatabaseCommand.java

```diff
@@ -31,43 +31,41 @@
  * }
  * </pre>
  */
-public final class DatabaseListener extends ListenerAdapter {
-
-    private static final Logger logger = LoggerFactory.getLogger(DatabaseListener.class);
-
+public class DatabaseCommand extends AbstractCommand {
+    private static final Logger logger = LoggerFactory.getLogger(DatabaseCommand.class);
     private final Database database;
 
     /**
      * Creates a new command listener, using the given database.
      *
      * @param database the database to store the key-value pairs in
      */
-    public DatabaseListener(Database database) {
+    public DatabaseCommand(Database database) {
+        super("Database Get & Set",
+                "This command can set or retrieve values from a key-object database.", true);
         this.database = database;
     }
 
     @Override
-    public void onMessageReceived(MessageReceivedEvent event) {
-        if (event.getAuthor().isBot()) {
-            return;
-        }
-        if (!event.isFromType(ChannelType.TEXT)) {
+    public void onSlashCommand(SlashCommandEvent event) {
+        if (event.getUser().isBot()) {
             return;
         }
-        if (event.isWebhookMessage()) {
+        if (event.getUser().isSystem()) {
             return;
         }
-        if (event.isFromType(ChannelType.PRIVATE)) {
+        if (event.getChannelType().equals(ChannelType.PRIVATE)) {
```

Comment:
i foresee that all those `if`-early-outs will be repeated on other commands as well. maybe we should move it a level up, into some utility methods or similar.

Reply:
I removed them since according to @Tais993, they do not make sense here, but we should consider adding some generic checks to the AbstractCommand class?

Reply:
What generic checks are you talking about?